exporter.go

```diff
@@ -351,105 +338,159 @@ func parseDogStatsDTagsToLabels(component string) map[string]string {
 	return labels
 }
 
-func (l *StatsDListener) handlePacket(packet []byte, e chan<- Events) {
-	lines := strings.Split(string(packet), "\n")
-	events := Events{}
-	for _, line := range lines {
-		if line == "" {
-			continue
-		}
+func lineToEvents(line string) (events Events) {
```

Comment:
Please remove the named return variable here as it adds stutter to the function signature and use explicit returns below. It's not a go idiom to use naked returns in such long functions.

> Named returned values should only be used on public funcs and methods
when it contributes to the documentation.

https://go-review.googlesource.com/c/20024/